library/proc_macro/src/bridge/mod.rs

```diff
@@ -176,7 +176,7 @@ macro_rules! with_api {
 }
 
 // FIXME(eddyb) this calls `encode` for each argument, but in reverse,
-// to avoid borrow conflicts from borrows started by `&mut` arguments.
+// to match the ordering in `reverse_decode`.
```

Comment:
I wrote a big comment I just had to remove, heh, until I figured out what was meant by this change.

So yeah it *is* about the borrow-checker but only directly so for `reverse_decode` and then `reverse_encode` has to play along.

I would still maybe add "(which is forced by borrowck)" or something at the end of the comment just to not make it look like the order is arbitrary.

Reply:
"to match the ordering in `reverse_decode`" was precisely my attempt to make it clear that the order is not arbitrary. I'm happy to hear alternative wording suggestions.